tests/ui_tests/test_admin_page.py

```diff
@@ -485,72 +487,67 @@ def test_delete_policy_from_policies_page(self, app_fs):
-
-        with allure.step(f"Create test role for service {service.name}"):
-            sdk_client_fs.role_create(
-                name=self.custom_role_name,
-                display_name=self.custom_role_name,
-                child=[{'id': sdk_client_fs.role(name='View service configurations').id}],
-            )
-        policies_page = AdminPoliciesPage(app_fs.driver, app_fs.adcm.url).open()
-        policies_page.create_policy(
-            policy_name=self.custom_policy.name,
-            description=self.custom_policy.description,
-            role=self.custom_policy.role,
-            users=self.custom_policy.users,
-            services=service.name,
-            parent=cluster.name,
-        )
-        self.check_custom_policy(policies_page)
-
-    def test_check_policy_popup_for_provider(self, sdk_client_fs, app_fs, create_cluster_with_component):
-        """Test creating policy for provider"""
+    @pytest.mark.parametrize(
+        ("clusters", "services", "providers", "hosts", "parents", "role_name"),
+        [
+            (CLUSTER_NAME, None, None, None, None, 'View cluster configurations'),
+            (None, SERVICE_NAME, None, None, CLUSTER_NAME, 'View service configurations'),
+            (None, None, PROVIDER_NAME, None, None, 'View provider configurations'),
+            (None, None, None, HOST_NAME, None, 'View host configurations'),
+            (None, SERVICE_NAME, None, None, CLUSTER_NAME, 'View component configurations'),
+            (CLUSTER_NAME, None, None, None, None, 'View cluster configurations, View service configurations'),
+            (
+                None,
+                SERVICE_NAME,
+                None,
+                None,
+                CLUSTER_NAME,
+                'View cluster configurations, View service configurations, View component configurations, '
+                'View host configurations',
+            ),
+            (None, None, PROVIDER_NAME, None, None, 'View provider configurations, View host configurations'),
+            (None, None, None, HOST_NAME, None, 'View provider configurations, View host configurations'),
+        ],
+    )
+    def test_check_policy_popup_for_entities(
+        self,
+        sdk_client_fs,
+        app_fs,
+        create_cluster_with_component,
+        clusters,
+        services,
+        providers,
+        hosts,
+        parents,
+        role_name,
+    ):
+        """Test creating policy"""
 
-        _, _, _, provider = create_cluster_with_component
         self.custom_policy.role = self.custom_role_name
-        self.custom_policy.objects = provider.name
-
-        with allure.step(f"Create test role for provider {provider.name}"):
+        if clusters:
+            self.custom_policy.objects = clusters
+        if services:
+            self.custom_policy.objects = services
+        if providers:
+            self.custom_policy.objects = providers
+        if hosts:
+            self.custom_policy.objects = hosts
```

Comment:
It may be like that or `clusters or services or providers or hosts` if not present is equal to None.
Maybe move this if's to some `sanitize_parametrization` function? It is a bit confusing to see such a tree inside the test body.

Reply:
yes, that much better